recipes/fs-truncate-fd-deprecation/src/workflow.ts

```diff
@@ -0,0 +1,313 @@
+  if (!hasChanges) return null;
+
+  return rootNode.commitEdits(edits);
+}
+
+/**
+ * Update import and require statements to replace truncate functions with ftruncate
+ */
+function updateImportsAndRequires(root: SgRoot<Js>, usedTruncate: boolean, usedTruncateSync: boolean, edits: Edit[]): void {
+  // @ts-ignore - ast-grep types are not fully compatible with JSSG types
+  const importStatements = getNodeImportStatements(root, 'fs');
+  // @ts-ignore - ast-grep types are not fully compatible with JSSG types
+  const requireStatements = getNodeRequireCalls(root, 'fs');
+
+  // Update import and require statements
+  for (const statement of [...importStatements, ...requireStatements]) {
+    let text = statement.text();
+    let updated = false;
+
+    if (usedTruncate && text.includes("truncate") && !text.includes("ftruncate")) {
+      text = text.replace(/\btruncate\b/g, "ftruncate");
+      updated = true;
+    }
+
+    if (usedTruncateSync && text.includes("truncateSync") && !text.includes("ftruncateSync")) {
+      text = text.replace(/\btruncateSync\b/g, "ftruncateSync");
+      updated = true;
+    }
+
+    if (updated) {
+      edits.push(statement.replace(text));
+    }
+  }
+}
+
+/**
+ * Check if a call expression is from a destructured fs import/require
+ */
+function isFromFsModule(call: SgNode<Js>, root: SgRoot<Js>): boolean {
+  // @ts-ignore - ast-grep types are not fully compatible with JSSG types
+  const importStatements = getNodeImportStatements(root, 'fs');
+  // @ts-ignore - ast-grep types are not fully compatible with JSSG types
+  const requireStatements = getNodeRequireCalls(root, 'fs');
+
+  // Get the function name being called (truncate or truncateSync)
+  const callExpression = call.child(0);
+  const functionName = callExpression?.text();
+  if (!functionName) return false;
+
+  // Check if this function name appears in any fs import/require destructuring
+  for (const statement of [...importStatements, ...requireStatements]) {
+    const text = statement.text();
+    if (text.includes("{") && text.includes(functionName)) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+/**
+ * Helper function to determine if a parameter is likely a file descriptor
+ * rather than a file path string.
+ * @param param The parameter to check (e.g., 'fd').
+ * @param rootNode The root node of the AST to search within.
+ */
+function isLikelyFileDescriptor(param: string, rootNode: SgNode<Js>): boolean {
+  // Check if it's a numeric literal
+  if (/^\d+$/.test(param.trim())) return true;
+
+  // Check if it's obviously a string literal (path)
+  if (/^['"`]/.test(param.trim())) return false;
+
+  // Check if it's assigned from fs.openSync or openSync
+  if (isAssignedFromOpenSync(param, rootNode)) return true;
+
+  // Check if it's used inside a callback context from fs.open
+  if (isInCallbackContext(param, rootNode)) return true;
+
+  // For other cases, be conservative - don't transform unless we're sure
+  return false;
+}
```

Comment:
nit: this can be simplified slightly by merging one or two if conditions

Reply:
Yeah you had right but written like that it's verbose and make easy to read/understand

Reply:
At least merge the last two using "OR", you can still add comments, it'll just run a bit faster since there are less computations